crates/miden-objects/src/account/auth.rs

```diff
@@ -303,10 +303,13 @@ impl Signature {
 
     /// Converts this signature to a sequence of field elements in the format expected by the
     /// native verification procedure in the VM.
-    pub fn to_prepared_signature(&self) -> Vec<Felt> {
+    pub fn to_prepared_signature(&self, msg: Word) -> Vec<Felt> {
+        // TODO: the `expect()` should be changed to an error; but that will be a part of a bigger
+        // refactoring
         match self {
             Signature::RpoFalcon512(sig) => prepare_rpo_falcon512_signature(sig),
-            Signature::EcdsaK256Keccak(sig) => prepare_ecdsa_k256_keccak_signature(sig),
+            Signature::EcdsaK256Keccak(sig) => miden_stdlib::prepare_ecdsa_signature(msg, sig)
```

Comment:
```suggestion
            Signature::EcdsaK256Keccak(sig) => prepare_ecdsa_k256_keccak_signature(msg, sig)
```

Related to my comment here: https://github.com/0xMiden/miden-base/pull/2074#issuecomment-3502068125
I think we also add this in reverse order:

```rust
/// Prepares an ECDSA K256 Keccak signature for verification inside the Miden VM.
fn prepare_ecdsa_k256_keccak_signature(
    msg: Word,
    sig: &ecdsa_k256_keccak::Signature,
) -> Vec<Felt> {
    // have the prepare signature function from miden stdlib in the reverse order
    let mut prepared_signature =
        miden_stdlib::prepare_ecdsa_signature(msg, sig).expect("should prepare ecdsa signature");
    prepared_signature.reverse();
    prepared_signature
}
```